includes/index_entity.inc

```diff
@@ -179,6 +179,25 @@ class SearchApiIndex extends Entity {
     parent::__construct($values, $entity_type);
   }
 
+  /**
+   * Checks if the entity has a certain exportable status.
+   *
+   * @param $status
+   *   A status constant, i.e. one of ENTITY_CUSTOM, ENTITY_IN_CODE,
+   *   ENTITY_OVERRIDDEN or ENTITY_FIXED.
+   *
+   * @return bool
+   *   For exportable entities TRUE if the entity has the status, else FALSE.
+   *   In case the entity is not exportable, NULL is returned.
+   *
+   * @see entity_has_status()
+   */
+  public function hasStatus($status) {
+    if (!empty($this->entityInfo['exportable'])) {
+      return isset($this->{$this->statusKey}) && ($this->{$this->statusKey} & $status) == $status;
```

Comment:
I dont have a complete answer, but although this may fix the error, the code is not really applicable in Backdrop since:
- we dont have the concept of exportable entities
- entities dont have a statuskey
- entities dont have a status

Reply:
I agree with you.
The concept of exportable entities is not in backdrop entity core - it's in entity plus in class EntityPlusControllerExportable, and is subsequently used by search api on line 418 of search_api.module

statusKey is also found in entity plus in EntityPlusControllerExportable, line 717 of entity_plus.controller.inc.

The confusion I have is if this should go in entity_plus or in search_api. As to the entity->status it's not part of entity core but is again defined in entity plus as a component of exportable entities, and referenced in search_api.

At this time search_api depends on entity_plus for a good deal of it's functionality, so that's my issue and question.
I have incorporated this proposed change and the pull_requests to entity_plus on my dev machine and it's working perfectly.
#https://github.com/backdrop-contrib/entity_plus/pull/31
#https://github.com/backdrop-contrib/entity_plus/pull/30
(one more pull request needs to be submitted to the entity_plus project for all this to work, I'm just waiting for the other two to be reviewed)
Despite the fact that in backdrop core entity module entities do not have a status, search indexes and search server objects do. (scroll to the bottom to verify)

SearchApiServer Object
(
    [id] => 1
    [name] => test1
    [machine_name] => test1
    [description] => 
    [class] => search_api_db_service
    [options] => Array
        (
            [database] => default:default
            [min_chars] => 1
            [partial_matches] => 1
            [indexes] => Array
                (
                    [testindex1] => Array
                        (
                            [nid] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => nid
                                    [type] => integer
                                    [boost] => 1.0
                                )

                            [vid] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => vid
                                    [type] => integer
                                    [boost] => 1.0
                                )

                            [is_new] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => is_new
                                    [type] => boolean
                                    [boost] => 1.0
                                )

                            [type] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => type
                                    [type] => string
                                    [boost] => 1.0
                                )

                            [title] => Array
                                (
                                    [table] => search_api_db_testindex1_text
                                    [type] => text
                                    [boost] => 1.0
                                )

                            [language] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => language
                                    [type] => string
                                    [boost] => 1.0
                                )

                            [url] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => url
                                    [type] => uri
                                    [boost] => 1.0
                                )

                            [edit_url] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => edit_url
                                    [type] => uri
                                    [boost] => 1.0
                                )

                            [status] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => status
                                    [type] => integer
                                    [boost] => 1.0
                                )

                            [promote] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => promote
                                    [type] => boolean
                                    [boost] => 1.0
                                )

                            [sticky] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => sticky
                                    [type] => boolean
                                    [boost] => 1.0
                                )

                            [created] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => created
                                    [type] => date
                                    [boost] => 1.0
                                )

                            [changed] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => changed
                                    [type] => date
                                    [boost] => 1.0
                                )

                            [author] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => author
                                    [type] => integer
                                    [boost] => 1.0
                                )

                            [source] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => source
                                    [type] => integer
                                    [boost] => 1.0
                                )

                            [log] => Array
                                (
                                    [table] => search_api_db_testindex1_text
                                    [type] => text
                                    [boost] => 1.0
                                )

                            [revision] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => revision
                                    [type] => boolean
                                    [boost] => 1.0
                                )

                            [field_tags] => Array
                                (
                                    [table] => search_api_db_testindex1_field_tags
                                    [column] => value
                                    [type] => list
                                    [boost] => 1.0
                                )

                            [search_api_language] => Array
                                (
                                    [table] => search_api_db_testindex1
                                    [column] => search_api_language
                                    [type] => string
                                    [boost] => 1.0
                                )

                        )

                )

        )

    [enabled] => 1
    [proxy:protected] => 
    [is_active_revision] => 1
    [status] => 1
    [module] => 
)